servatrice/src/serversocketinterface.cpp

```diff
@@ -1257,24 +1263,75 @@ Response::ResponseCode AbstractServerSocketInterface::cmdAccountEdit(const Comma
     if (authState != PasswordRight)
         return Response::RespFunctionNotAllowed;
 
-    QString realName = QString::fromStdString(cmd.real_name());
-    QString emailAddress = QString::fromStdString(cmd.email());
-    QString country = QString::fromStdString(cmd.country());
-
+    bool checkedPassword = false;
     QString userName = QString::fromStdString(userInfo->name());
+    if (cmd.has_password_check()) {
+        QString password = QString::fromStdString(cmd.password_check());
+        QString clientId = QString::fromStdString(userInfo->clientid());
+        QString reasonStr{};
+        int secondsLeft{};
+        AuthenticationResult checkStatus =
+            databaseInterface->checkUserPassword(this, userName, password, clientId, reasonStr, secondsLeft, true);
+        if (checkStatus == PasswordRight) {
+            checkedPassword = true;
+        } else {
+            // the user already logged in with this info, the only change is their password
+            return Response::RespWrongPassword;
+        }
+    }
+    QString queryText = "update {prefix}_users set ";
+    bool changes = false;
+    if (cmd.has_real_name()) {
+        queryText += "realname=:realName, ";
```

Comment:
queries should not be built up like a standard string. If you dont have a value you need, use a null or default. I really don't like this.

Reply:
well, please explain to me how you'd prefer it then.

Reply:
I'm thinking two queries, one to get the current values and one to update. If the user didn't override, then place the current value in the variable. Else use the new value. 

I really don't like query builders with raw strings, it's begging for an issue

Reply:
We can also look into a query builder, maybe like https://github.com/MasterAler/qsqlbuilder ?

Reply:
Spoke with a friend, they recommended using a QStringList to build up the parts, then a `list.join(',')` so we don't have to manually modify strings. Also, moving the query build component to the bottom could be useful so we can see the full query without having to scout around the file.  `UPDATE table SET {list.join(',')} WHERE conditionals` as a basic approach